rasa/core/policies/rule_policy.py

```diff
@@ -772,7 +787,7 @@ def train(
         Returns:
             The resource which can be used to load the trained policy.
         """
-        self._validate_against_domain(domain)
+        self.raise_if_incompatible_with_domain(self.config, domain)
```

Comment:
we can drop this one then as we already have this in the `ConfigValidator`, right?

Reply:
I'd keep it because it's a quick check and someone might want to play around with this component as standalone and then it might be handy, wdyt?

Reply:
I'd say they're on their own risk 😁  Both options are fine for me 👍🏻